### ISSUE TITLE
Data Center Model FMU initialization error

### ISSUE THREAD

**Issue by AvisekNaug** [lbl-srg/modelica-buildings#3169]
**TLDR**

Error in fmu.initialize() step related to `Model error: Argument of log(ahu.cooCoi.dryWetCalcs.TAirInDewPoi) was 0 should be > 0` in `Buildings.Applications.DataCenters.ChillerCooled.Examples.IntegratedPrimaryLoadSideEconomizer.mo`  when using FMU in `ME&CS` mode compiled using Open Modelica simulator

**Detailed Issue**

I have compiled the `Buildings.Applications.DataCenters.ChillerCooled.Examples.IntegratedPrimaryLoadSideEconomizer.mo` file using the OMEdit GUI with default settings. My overall goal is to use the PyFMI Cosimulation Mode "do_step" method to simulate the data center step-by-step. 

When I try to initialize() the FMU using pyFMI, it generates an error which I cannot figure out. I have not changed anything in the above modelica file or the files it uses.

**System Specs**
System: Linux Ubuntu
Python : 3.9.15; 
pyfmi :2.9.8   ( py39h2ae25f5_1  conda-forge)
building library used: [Modelica Buildings Library](https://simulationresearch.lbl.gov/modelica/download.html) zip file
OpenModelica being used:
```
OMEdit - OpenModelica Connection Editor
Connected to OpenModelica 1.21.0~dev-40-g75ffc88
Connected to OMSimulator unknown-linux
```

**I have included the step by step methods to recreate the error**

1. Compiled the model using the Export FMU option in OMEdit with the following options for FMI export: [Options](https://imgur.com/a/Td6VSm8); (Chaning solver form Euler to CVODE gives the same error.)
2. Set up a conda env with Python 3.9.15
3. Conda Install PyFMI from [here](https://anaconda.org/conda-forge/pyfmi) using `conda install -c conda-forge pyfmi`
4. Then run the following steps to load the FMU and then error happens at the initialize step below.
```python
from pyfmi import load_fmu

fmu_path = 'Buildings.Applications.DataCenters.ChillerCooled.Examples.IntegratedPrimaryLoadSideEconomizer.fmu'
fmu = load_fmu(fmu_path, log_level=5)

start_hour = 0
final_hour = 30
start_time = 3600*start_hour
final_time = 3600*final_hour

fmu.reset()
fmu.initialize(start_time=start_time,stop_time=final_time)  # Error happens at this stage

# More code for running set-by-step simulation using fmu.do_step(...) and collect some of the values of certain variables using fmu.get() method: fmu.get('roo.TRooAir')
```
Error at the terminal
```bash
assert            | debug   | Model error: Argument of log(ahu.cooCoi.dryWetCalcs.TAirInDewPoi) was 0 should be > 0
stdout            | warning | Non-Linear Solver try to handle a problem with a called assert.
assert            | debug   | Model error: Argument of log(ahu.cooCoi.dryWetCalcs.TAirInDewPoi) was 0 should be > 0
assert            | debug   | Model error: Argument of log(ahu.cooCoi.dryWetCalcs.TAirInDewPoi) was 0 should be > 0
assert            | debug   | Model error: Argument of log(ahu.cooCoi.dryWetCalcs.TAirInDewPoi) was 0 should be > 0
assert            | debug   | Solving non-linear system 2617 failed at time=0.
|                 | |       | For more information please use -lv LOG_NLS.
assert            | error   | Failed to solve the initialization problem with global homotopy with equidistant step size.
assert            | debug   | Unable to solve initialization problem.
Traceback (most recent call last):
  File "/home/janedoe/john/dt/dev/run_fmu.py", line 21, in <module>
    fmu.initialize(start_time=start_time,stop_time=final_time)
  File "src/pyfmi/fmi.pyx", line 4798, in pyfmi.fmi.FMUModelBase2.initialize
  File "src/pyfmi/fmi.pyx", line 4793, in pyfmi.fmi.FMUModelBase2.initialize
  File "src/pyfmi/fmi.pyx", line 4707, in pyfmi.fmi.FMUModelBase2.exit_initialization_mode
pyfmi.fmi.FMUException: Exit Initialize returned with an error. Enable logging for more information, (load_fmu(..., log_level=4)).
```
The corresponding log file shows the following errors relevant to the initialization stage. Full log file is in [this](https://pastebin.com/5gWaNp5Q) pastebin
```bash
FMIL: module = FMILIB, log level = 5: Successfully loaded all the interface functions
FMIL: module = Model, log level = 4: [logFmi2Call][FMU status:OK] fmi2Instantiate: GUID={3016e3c6-8897-4db2-9e10-2f43eede012f}
FMIL: module = Model, log level = 4: [logFmi2Call][FMU status:OK] fmi2Reset
FMIL: module = FMICAPI, log level = 5: Calling fmi2SetupExperiment
FMIL: module = Model, log level = 4: [logFmi2Call][FMU status:OK] fmi2SetupExperiment: toleranceDefined=1 tolerance=1e-06 startTime=0 stopTimeDefined=0 stopTime=108000
FMIL: module = FMICAPI, log level = 5: Calling fmi2EnterInitializationMode
FMIL: module = Model, log level = 4: [logFmi2Call][FMU status:OK] fmi2EnterInitializationMode...
FMIL: module = FMICAPI, log level = 5: Calling fmi2ExitInitializationMode
FMIL: module = Model, log level = 4: [logFmi2Call][FMU status:OK] fmi2ExitInitializationMode...
FMIL: module = Model, log level = 3: [logStatusWarning][FMU status:Warning] The following assertion has been violated during initialization at time 0.000000
FMIL: module = Model, log level = 3: [logStatusWarning][FMU status:Warning] The following assertion has been violated during initialization at time 0.000000
FMIL: module = Model, log level = 3: [logStatusWarning][FMU status:Warning] The following assertion has been violated during initialization at time 0.000000
FMIL: module = Model, log level = 3: [logStatusWarning][FMU status:Warning] The following assertion has been violated during initialization at time 0.000000
FMIL: module = Model, log level = 2: [logFmi2Call][FMU status:Error] fmi2EnterInitializationMode: terminated by an assertion.
FMIL: module = Model, log level = 4: [logFmi2Call][FMU status:OK] fmi2ExitInitializationMode: succeed
FMIL: module = Model, log level = 4: [logFmi2Call][FMU status:OK] fmi2FreeInstance
FMIL: module = FMILIB, log level = 5: Releasing FMU CAPI interface
FMIL: module = FMICAPI, log level = 5: Successfully unloaded FMU binary
FMIL: module = FMILIB, log level = 5: Releasing allocated library resources
FMIL: module = JMPRT, log level = 5: Removing /tmp/janedoe/JModelica.org/jm_tmp3xtuq3g8
```

PS: 
If I do not initialize the fmu, then I cannot get any values using the fmu.get() method. It generates cannot get RealValue error.

**Comment by AvisekNaug** [comment]
I do want to mention that compiling and running the same model in Version 8.1.3 using Jmodelica works without error in the same PyFMi.

**Comment by AvisekNaug** [comment]
Thanks for the prompt response! I will try to work through each suggestion and let you know which ones work.
